source/app/models/game_logic.rb

```diff
@@ -0,0 +1,43 @@
+class GameLogic
+
+# RULES = {
+#   "rock" (1) => "scissor", (3)
+#   "paper" (2) => "rock", (1)
+#   "scissor"(3) => "paper" (2)
+# }
+
+RULES = {
+  1 => 3,
+  2 => 1,
+  3 => 2
+  }
+
+
+  def initialize(player_1_id, player_2_id, player_1_token_id, player_2_token_id)
+    @player_1_id = player_1_id
+    @player_2_id = player_2_id
+    @player_1_token_id = player_1_token_id
+    @player_2_token_id = player_2_token_id
+  end
+
+  def outcome?
```

Comment:
instead of outcome, with all this logic.... why not just make two methods...

who_won?
   {winner: winner_id, winner_token: winner_token_id}
end

who_lost?
 {loser: loser_id, loser_token: loser_token_id}
end